packages/shared-ux/chrome/navigation/src/ui/navigation_bucket.tsx

```diff
@@ -6,36 +6,88 @@
  * Side Public License, v 1.
  */
 
-import { EuiCollapsibleNavGroup, EuiIcon, EuiSideNav, EuiText } from '@elastic/eui';
+import {
+  EuiCollapsibleNavGroup,
+  EuiIcon,
+  EuiSideNav,
+  EuiSideNavItemType,
+  EuiText,
+} from '@elastic/eui';
 import React from 'react';
-import { NavigationBucketProps } from '../../types';
+import { ChromeNavigationNodeViewModel } from '../../types';
+import type { BasePathService, NavigateToUrlFn } from '../../types/internal';
 import { useNavigation } from '../services';
 import { navigationStyles as styles } from '../styles';
 
-export const NavigationBucket = (opts: NavigationBucketProps) => {
-  const { id, items, activeNavItemId, ...props } = opts;
-  const { navIsOpen } = useNavigation();
+const navigationNodeToEuiItem = (
```

Comment:
This is where we convert our `ChromeNavigationNodeViewModel` to a `EuiSideNavItemType` to render the jsx.